src/TorchSharp/Tensor/Storage.cs

```diff
@@ -0,0 +1,522 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using System.Diagnostics;
+using System.Linq;
+using System.Runtime.CompilerServices;
+
+namespace TorchSharp
+{
+    public static partial class torch
+    {
+        /// <summary>
+        /// A torch.Storage is a contiguous, one-dimensional array of a single data type.
+        /// Every tensor has a corresponding storage of the same data type.
+        /// </summary>
+        public abstract class Storage
+        {
+            internal Storage()
```

Comment:
>internal [](http://example.com/codeflow?start=12&length=8)

should we have the `internal` as `protected` if these are going to be used by the subclasses only?